### PR TITLE
Documentation for e2e under wayland

### PR DESCRIPTION
**Pull request by Luis-omega**

### CODE REVIEW

File: doc/faq.md [hunk 1]
```diff
@@ -45,6 +46,12 @@ Ensure your transaction's validity range does not go over `SafeZone` slots of th
 
 Haskell's `aeson` library encodes long integers as JSON numbers, which leads to numeric truncation on decoder side if JS `Number` is used. Unfortunately, `purescript-argonaut` does not allow to use another type, because the truncation happens during `JSON.parse` call. `purescript-aeson` is our custom solution that bypasses this limitation by storing numbers as strings. It exposes a very similar API.
 
+## Environment-related
+
+### Q: Why I can't open the browser used for e2e?
```
[the remark this answers is not in the document]
Comment (klntsky) [reply doc/faq.md:51]: (put an error message here so that it's searchable)

[the remark this answers is not in the document]
Comment (Luis-omega) [reply doc/faq.md:51]: I only get a classic `Couldn't find X display server` error message once by luck. After that, I only get an environment-specific and cryptic chromium error message that users definitively won't relate to using `Wayland` and `CTL e2e`, I don't think that would make this more searchable.

[the remark this answers is not in the document]
Comment (Luis-omega) [reply doc/faq.md:51]: The log of the error looks like this : 

```
10 silly lifecycle cardano-transaction-lib@3.0.0~e2e-browser: Args: [
10 silly lifecycle   '-c',
10 silly lifecycle   "source ./test/e2e.env && spago run --main Test.Ctl.E2E -a 'e2e-test browser'"
10 silly lifecycle ]
11 silly lifecycle cardano-transaction-lib@3.0.0~e2e-browser: Returned: code: 1  signal: null
12 info lifecycle cardano-transaction-lib@3.0.0~e2e-browser: Failed to exec e2e-browser script
13 verbose stack Error: cardano-transaction-lib@3.0.0 e2e-browser: `source ./test/e2e.env && spago run --main Test.Ctl.E2E -a 'e2e-test browser'`
13 verbose stack Exit status 1
13 verbose stack     at EventEmitter.<anonymous> (/nix/store/lrvrir70n3966jybpjqw91smhcwlyn00-nodejs-14.20.0/lib/node_modules/npm/node_modules/npm-lifecycle/index.js:332:16)
13 verbose stack     at EventEmitter.emit (events.js:400:28)
13 verbose stack     at ChildProcess.<anonymous> (/nix/store/lrvrir70n3966jybpjqw91smhcwlyn00-nodejs-14.20.0/lib/node_modules/npm/node_modules/npm-lifecycle/lib/spawn.js:55:14)
13 verbose stack     at ChildProcess.emit (events.js:400:28)
13 verbose stack     at maybeClose (internal/child_process.js:1088:16)
13 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:296:5)
14 verbose pkgid cardano-transaction-lib@3.0.0
```

[the remark this answers is not in the document]
Comment (klntsky) [reply doc/faq.md]: Yep, please put this message in the FAQ.

Use expandable spoilers like here https://gist.github.com/jbsulli/03df3cdce94ee97937ebda0ffef28287#demo to hide logs

[the remark this answers is not in the document]
Comment (klntsky) [reply doc/faq.md]: The user may miss the FAQ, but google will send them here if there's a message text

**Comment by Luis-omega** [comment]
There's an error about trailing whitespace introduced by the use of the tag `summary`.

```
Some source files have trailing spaces:
CHANGELOG.md:- [Ogmios](https://github.com/mlabs-haskell/ogmios) - v5.5.7
make: *** [Makefile:34: check-whitespace] Error 1
```

```
<details>
  <summary>Spoiler warning</summary>
  
  Spoiler text. Note that it's important to have a space after the summary tag. You should be able to write any markdown you want inside the `<details>` tag... just make sure you close `<details>` afterward.
  
  ```javascript
  console.log("I'm a code block!");
  ```
  
</details>
```

File: doc/faq.md [hunk 2]
```diff
@@ -64,6 +64,39 @@ Haskell's `aeson` library encodes long integers as JSON numbers, which leads to
 
 ### Q: I use wayland, the E2E browser fails on startup
 
+We are aware of two error messages that can be show to you if you are using wayland.
+
+<details>
+  <summary>You can get something like this if you try to open the e2e browser</summary>
+
+  10 silly lifecycle cardano-transaction-lib@3.0.0~e2e-browser: Args: [
```
Comment (klntsky) [inline doc/faq.md]: This does not render properly, use code blocks inside.

https://github.com/Plutonomicon/cardano-transaction-lib/blob/4c82c3870a10fffa4e9f27e69fc3bd7017fb43aa/doc/faq.md

**Review by klntsky — CHANGES_REQUESTED** [review]

File: doc/faq.md (hunk 2 above)
Comment (Luis-omega) [reply doc/faq.md]: it's already fixed
https://github.com/Plutonomicon/cardano-transaction-lib/blob/cf0ade4f9d8daf3ebb2c7bcd08ea4760076cd6a5/doc/faq.md

Comment (klntsky) [reply doc/faq.md]: Thanks, could you please remove the log levels too (`silly lifecycle` and friends)

**Review by klntsky — APPROVED** [review]

**Pull request merged**